contracts/test/Verifiers.ts

```diff
@@ -0,0 +1,184 @@
+      range_check_snark_proof: BytesLike;
+      grand_sums_batch_proof: BytesLike;
+      total_balances: BigNumber[];
+    };
+
+    beforeEach(async () => {
+      const deploymentInfo = await loadFixture(deployVerifyingFixture);
+      verifyingKey = deploymentInfo.verifyingKey;
+      commitmentCalldata = deploymentInfo.commitmentCalldata;
+
+      // Deploy GrandSumVerifier contract
+      grandSumVerifier = await ethers.deployContract(
+        "src/GrandsumVerifier.sol:GrandsumVerifier"
+      ) as GrandsumVerifier;
+    });
+
+    it("should verify grandsum proof", async () => {
+      // Concatenates the snark proof and the grand sum proof
+      let snarkProofArray = ethers.utils.arrayify(commitmentCalldata.range_check_snark_proof);
+      let grandSumProofArray = ethers.utils.arrayify(commitmentCalldata.grand_sums_batch_proof);
+      let totalBalances = commitmentCalldata.total_balances;
+
+      // The first 64 bytes of the snark proof represent a commitment to the corresponding username polynomial
+      // Starting from the next 64 bytes, each set of 64 bytes represents commitments corresponding to the total sum of balances
+      let grandSumCommitments = snarkProofArray.slice(64, (64 + grandSumProofArray.length));
+
+      // The verifier iterates over points in the proofs while verifying them.
+      // The proofs look like:
+      //  i = 0                                       1                                               N                 
+      // [grand_sum_proof_p1_x, grand_sum_proof_p1_y, grand_sum_proof_p2_x, grand_sum_proof_p2_y, ... grand_sum_proof_pN_x, grand_sum_proof_pN_y, ...]
+      // [    snark_proof_p1_x,     snark_proof_p1_y,     snark_proof_p2_x,     snark_proof_p2_y, ...     snark_proof_pN_x,     snark_proof_pN_y, ...] 
+      //  Where `N` is the number of currencies
+      let proofs = ethers.utils.concat([grandSumProofArray, grandSumCommitments]);
+
+      expect(await grandSumVerifier.verifyProof(verifyingKey.address, proofs, totalBalances)).to.be.not.reverted;
+    });
+  });
+
+
+  describe("Inclusion Proof Verifier", () => {
+    let inclusionVerifier: InclusionVerifier;
+    let verifyingKey: Halo2VerifyingKey;
+    let inclusionProof: BytesLike;
+    let commitmentCalldata: {
+      range_check_snark_proof: BytesLike;
+      grand_sums_batch_proof: BytesLike;
+      total_balances: BigNumber[];
+    };
+    let challenges: [BigNumber, BigNumber, BigNumber, BigNumber];
+    let username: BytesLike;
+    let username_biguint: BigNumber;
+    let balance1: BigNumber;
+    let balance2: BigNumber;
+
+    beforeEach(async () => {
+      const deploymentInfo = await loadFixture(deployVerifyingFixture);
+      verifyingKey = deploymentInfo.verifyingKey;
+      commitmentCalldata = deploymentInfo.commitmentCalldata;
+
+      // InclusionVerifier requires BN256G2 contract for performing elliptic curve operations on G2 subgroup
+      // const bn256g2 = await deployBN256G2();
```

Comment:
This can be removed

Reply:
It removed at https://github.com/summa-dev/summa-solvency/commit/6db5fb7cc56192741c86d55ff8f59c9eb7dea05b